File: Asocioacion/php/cuartoEjercicio/Camion.php
<?php

//Importe de la Super Clase
require_once('Automotor.php');

class Camion extends Automotor{

    private int $numeroEjes;
    private int $capacidadCarga;

    public function __construct(string $marca, int $modelo, float $precio, int $numeroEjes, int $capacidadCarga){
        parent::__construct($marca, $modelo, $precio);

        $this->numeroEjes = $numeroEjes;
        $this->capacidadCarga = $capacidadCarga;

    }

    //Numero de ejes
    public function getNumeroEjes(){
        return $this->numeroEjes;
    }
    public function setNumeroEjes($numeroEjes){
        $this->numeroEjes = $numeroEjes;
    }

    //Capacidad de carga
    public function getCapacidadCarga(){
        return $this->capacidadCarga;
    }
    public function setCapacidadCarga($capacidadCarga){
        $this->capacidadCarga = $capacidadCarga;
    }

    //Mostrar los datos
    public function mostrarDatos(){
        echo '<h2>'.'<b>'.'CAMIONES'.'</b>'.'</h2>';        
        echo 'La marca del automotor es: '. $this->getMarca() .'<br>';
        echo 'El modelo del automotor es: '.$this->getModelo().'<br>';
        echo 'El precio del automotor es: '.$this->getPrecio().'<br>';
        echo 'El numero de ejes del camion son: '. $this->numeroEjes.'<br>';
        echo 'La capacidad de carga del camion es: '. $this->capacidadCarga.'<br>';
    }
}

?>

File: Asocioacion/python/cuartoEjercicio/Camion.py
#Importe de la clase padre
import Automotor as au

class Camion(au.Automotor):
    c = 0
    def __init__(self, marca, modelo, precio, numeroEjes, capacidadCarga):
        super().__init__(marca, modelo, precio)
        self.numeroEjes = numeroEjes
        self.capacidadCarga = capacidadCarga
       
    #Numero de ejes 
    def getNumeroEjes(self):
        return self.numeroEjes
    
    def setNumeroEjes(self, numeroEjes):
        self.numeroEjes = numeroEjes
        
    #Capacidad de carga
    def getCapacidadCarga(self):
        return self.capacidadCarga
    
    def setCapacidadCarga(self, capacidadCarga):
        self.capacidadCarga = capacidadCarga
        
    #Mostrar datos del camion
    def mostrarDatos(self):
        print('CAMIONES')
        print('-'*50)
        print(f'La marca del automotor es: {self.marca}')
        print(f'El modelo del automotor es: {self.modelo}')
        print(f'El precio del automotor es: {self.precio}')  
        print(f'El numero de ejes del camion son: {self.numeroEjes}')
        print(f'La capacidad de carga del camion es de: {self.capacidadCarga} toneladas')                